starter_code/src/movies.js

```diff
@@ -1,16 +1,68 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+
+function ratesAverage(array) {
+  let rates = array.reduce(function(accumulator, currentValue) {
+  
+    return accumulator + parseFloat(currentValue.rate);
+  }, 0);
+
+  return Number((rates / array.length).toFixed(2));
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
 
+// function dramaMoviesRate(array) {
+//   const isDrama = function(drama) {
+//     return array.genre === "Drama";
+//   };
+
+//   let dramaMovie = array.filter(isDrama).reduce(function(iterator, genre) {
+//     return iterator + parseFloat(genre.drama);
+//   });
+
+//   return (dramaMovie / array.length).toFixed(2);
+// }
+
+function dramaMoviesRate(movies) {
+  let dramaMovieCount = 0;
+  let dramaMovies = movies.reduce((rates, movie) => {
+    if (movies.genre.includes("Drama")) {
```

Comment:
Here you want to target the single movie, not movies.genre. 


It might make sense here to filter first all drama movies and then get the average using the reduce method. But your code works as well after changing the movie - movies on line 31.